test/Result/Result.Tests.cs

```diff
@@ -29,5 +29,35 @@ public void Failure()
             Assert.False(result2);
             Assert.True(result2.IsError);
         }
+
+        [Fact]
+        public async Task SuccessWithTask()
+        {
+            // Arrange
+            Result result1 = Result.Ok();
+            Task<Result> taskResult() => Task.FromResult(result1);
+
+            // Act
+            var result2 = await taskResult();
```

Comment:
Here we can just invoke the local function above:
`var result = await produceTaskWithSuccessfulResult();`